pkg/client/unversioned/horizontalpodautoscaler.go

```diff
@@ -40,12 +40,14 @@ type HorizontalPodAutoscalerInterface interface {
 
 // horizontalPodAutoscalers implements HorizontalPodAutoscalersNamespacer interface
 type horizontalPodAutoscalers struct {
-	client *ExtensionsClient
+	// Use RESTClient instead of more specific to allow reusing the type in different clients.
+	// TODO(piosz): change to AutoscalingClient once completely migrated to 'autoscaling' group.
+	client *RESTClient
```

Comment:
> To avoid duplication of the code. Alternatively I can create horizontalPodAutoscalersV1 which has the same logic but takes AutoscalingClient or implemented the solution you proposed with 'if' in each method to check which client is not nil.

Relaxing the constraint in this way makes it very easy to accidentally construct this using an incorrect type and not realize it until first touch at runtime.  The duplicated code that specific typing would produce is annoying, but it helps every consumer of this API use it correctly.  I'd rather have the duplication here and save everyone else from accidents.  That probably even ripples to two different implementations that satisfy the same interface to make the distinction easy to follow.